core/textinput/src/textinput/TerminalDisplayWin.cpp

```diff
@@ -35,13 +35,12 @@ namespace textinput {
         FILE_SHARE_READ | FILE_SHARE_WRITE, NULL, OPEN_EXISTING,
         FILE_ATTRIBUTE_NORMAL, NULL);
       ::GetConsoleMode(fOut, &fOldMode);
-
-      CONSOLE_SCREEN_BUFFER_INFO csbi;
-      ::GetConsoleScreenBufferInfo(fOut, &csbi);
-      fDefaultAttributes = csbi.wAttributes;
-      assert(fDefaultAttributes != 0 && "~TerminalDisplayWin broken");
     } else
       ::SetConsoleOutputCP(65001); // Force UTF-8 output
+    CONSOLE_SCREEN_BUFFER_INFO csbi;
+    ::GetConsoleScreenBufferInfo(fOut, &csbi);
+    fDefaultAttributes = csbi.wAttributes;
+    assert(fDefaultAttributes != 0 && "~TerminalDisplayWin broken");
```

Comment:
Could you please add parenthesis to the else clause? It's unclear if you wanted this to be part of the else or to be outside the if/else altogether.